PSOperations/RetryOperation.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  RetryOperation.swift
+//  PSOperations
+//
+//  Created by Bart Whiteley on 9/22/15.
+//
+
+
+public class RetryOperation: GroupOperation {
+    private let operationProducer: () -> Operation
+    private let shouldRetry: (errors:[NSError], retryCount:Int) -> Bool
+    private var numberOfRetries: Int = 0
+    
+    public convenience init(@autoclosure(escaping) operation: () -> Operation, maximumNumberOfRetries: Int = 5) {
+        self.init(operationProducer: operation, shouldRetry: { (_, retries) in return retries < maximumNumberOfRetries })
+    }
+    
+    public init(operationProducer: Void -> Operation, shouldRetry: (errors:[NSError], retryCount:Int) -> Bool) {
```

Comment:
I wonder if it would be more helpful to pass the `Operation` here into the `shouldRetry` closure, so you could look at more than just errors to determine if you should retry. I believe you can still access the errors from the Operation if you want them. What do you think?